### PR TITLE
Feature/notification

### PR DESCRIPTION
**Pull request by Brad123ghost**
### **Please test thoroughly, as there may be bugs due to the main merge.**

**Changes**
- Add UI for super/user notification
- Add notification system for normal users and super users
- Add ability to update notification to "read" or "assign" it
- Add token verify to authService
- Change returned payload of auth service
- Change the uid of notification to be user
- Add default value for guest submission
- Merge dev changes to branch

**Misc**
- change notification schema to fix error
- remove redundant code
- change `article `to `articles`
- fix various artefacts from main merge
- fix issues with front and backend build/test

**Testing**
- Unit test have not been written for this as of now

**To do**
- Need to add the admin side of updating notification, hopefully this will be done on the moderate/analyze view where the user can assign themselves

**Notes**
- Temporarily made it so when you submit article via backend URL it adds a notification to user as an "approved" notification for testing sake, viewing the notification should bring you to the article page and "read" it however this can change to just be dismiss and have it be viewable once it's been posted.

### CODE REVIEW

File: backend/src/main.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ import { Logger } from '@nestjs/common';
 async function bootstrap() {
   const app = await NestFactory.create(AppModule);
   const configService = app.get(ConfigService);
-  const port = configService.get<number>('PORT') || 5000;
+  const port = configService.get<number>('PORT') || 5001;
```
Comment (jackdar) [inline backend/src/main.ts:9]: dev branch is now using 8787

**Comment by Brad123ghost** [comment]
Please test thoroughly, as there may be bugs due to the main merge.

**Comment by Brad123ghost** [comment]
OMG finally all checks passed, notification service still has no unit test tho, who cares

**Comment by jackdar** [comment]
Sweet mate will test when home later this evening

File: frontend/src/app/hooks/useAuth.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+export interface AuthUser {
+  userId: string;
+  firstName: string;
+  lastName: string;
+  email: string;
+  role: string;
+}
+
+export function useAuth() {
```
Comment (jackdar) [inline frontend/src/app/hooks/useAuth.ts]: The useAuth function is specified in /src/contexts/auth-context.tsx where it uses AuthContext that implements all the auth state variables, it has been implemented twice and is causing some issues with logging in and out & state re-renders. Please combine into one useAuth hook.

File: frontend/src/app/types/notification/admin-notification.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+export interface AdminNotifcation {
```
Comment (jackdar) [inline frontend/src/app/types/notification/admin-notification.ts]: Small thing but other type defs have been moved out of app dir in ./src/types

File: frontend/src/app/types/notification/user-notification.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+export interface UserNotification {
```
Comment (jackdar) [inline frontend/src/app/types/notification/user-notification.ts]: See admin-notification.ts

File: frontend/src/components/submission/submission-form.tsx [hunk 5]
```diff
@@ -49,11 +49,12 @@ const PaperForm = () => {
   const onSubmit = async (data: z.infer<typeof schema>) => {
     try {
       const response = await fetch(
-        `${process.env.NEXT_PUBLIC_API_URL}/article/new`,
+        `${process.env.NEXT_PUBLIC_API_URL}/articles/new`,
         {
           method: 'POST',
           headers: {
             'Content-Type': 'application/json',
+            'Authorization': "Bearer " + localStorage.getItem("token")
```
Comment (jackdar) [inline frontend/src/components/submission/submission-form.tsx]: ./src/contexts/auth-context.tsx useAuth() exports token and can be used here

File: frontend/src/components/navbar.tsx [hunk 6]
```diff
@@ -1,9 +1,10 @@
 'use client';
 
-import { useAuth } from '@/hooks/use-auth';
+import { useAuth } from "../app/hooks/useAuth";
```
Comment (jackdar) [inline frontend/src/components/navbar.tsx]: please see ./src/app/hooks/useAuth.ts

**Review by jackdar — COMMENTED** [review]
Needs a few changes mainly doubled implementation of useAuth()

File: frontend/src/app/hooks/useAuth.ts (hunk 2 above)
Comment (Brad123ghost) [reply frontend/src/app/hooks/useAuth.ts]: File has been deleted, artefact of merge

File: frontend/src/components/navbar.tsx (hunk 6 above)
Comment (Brad123ghost) [reply frontend/src/components/navbar.tsx]: updated

File: frontend/src/components/submission/submission-form.tsx (hunk 5 above)
Comment (Brad123ghost) [reply frontend/src/components/submission/submission-form.tsx]: added useAuth, changed to `Bearer ${token}`

File: frontend/src/app/types/notification/user-notification.ts (hunk 4 above)
Comment (Brad123ghost) [reply frontend/src/app/types/notification/user-notification.ts]: moved

File: frontend/src/app/types/notification/admin-notification.ts (hunk 3 above)
Comment (Brad123ghost) [reply frontend/src/app/types/notification/admin-notification.ts]: moved

**Review by jackdar — APPROVED** [review]
lgtm 👍

**Pull request merged**